### PR TITLE
Zookeeper as a full first-party plugin that can be turned off

### LINKED ISSUE #12476: Zookeeper does not update when the user closes the pane after sending the request, and pane remains closed till completion
As discussed in Slack, you discovered through tracing that the case where the pane is opened completely after the turn is distinct, because it does not trigger a disconnect detected by Zookeeper. That can be addressed separately.

_Originally posted by @franknoirot in https://github.com/KittyCAD/modeling-app/pull/12472#pullrequestreview-4715913086_

### PR DESCRIPTION
**Pull request by pierremtb**
Closes #12476. Part of #12583.

The project-session signal introduced in #12885 is the non-React lifetime boundary.

## Why

The Zookeeper actor, WebSocket, prompt queue, reconnect logic, and file-edit bookkeeping used to be owned by the React pane. Closing the pane unmounted those owners, so an active prompt stopped and reopening required a new connection.

This moves session ownership to the Zookeeper registry runtime. There is no hidden React host or portal: pane visibility now controls only the UI, while disabling the plugin tears down the complete live Zookeeper runtime. The persisted conversation mapping remains available to the copied-project identity repair introduced in #13607; that storage utility does not start Zookeeper.

## What changed

- While the plugin is enabled, a registry runtime owns one controller for the active project session. It starts as soon as auth and the project editor are ready, independently of whether the pane is open, and updates auth in place.
- The runtime accepts the project ID from project-scoped settings only after the settings and project paths agree, then freezes it into the controller. Project path, project ID, and executing-editor identity define the session lifetime; the controller does not observe later settings mutations.
- A non-React controller owns the manager actor, prompt queue, reconnect and conversation persistence, billing transitions, clear-chat, interrupted-turn resume, and file-edit workers. Store operations are ordered globally and drained during disposal.
- File writes and undo-history recording moved from React hooks into serialized, disposable workers with project/editor epoch checks. Snapshot capture is all-or-nothing and falls back to the streamed patch if a file cannot be read safely.
- The pane and wrapper are presentation adapters over the controller. While a hidden session is processing, the pane toggle replaces the sparkles icon with a spinner.

## Review guide

The raw diff is larger than the behavioral change because GitHub cannot recognize hooks and callbacks that were split and converted into non-React classes. Most of the old pane/wrapper orchestration is removed and reappears behind focused interfaces.

1. src/lib/zookeeper/registry/index.ts — start at the plugin boundary. The plugin contributes the pane runtime directly, so registry enablement activates the runtime, while awaited plugin or registry disposal drains it.
2. src/lib/zookeeper/registry/runtime.tsx — the central lifetime policy. It deliberately has no layout dependency. Settings IDs are accepted only when their project path matches the active project, so route transitions cannot mix sessions. Same-project republishes and temporary readiness gaps retain the controller; logout, project-path or project-ID changes, a real editor replacement, or plugin disposal stop it. Replacements sharing a path or editor wait for the previous asynchronous teardown, including rapid disable/re-enable transitions.
3. src/lib/zookeeper/registry/controller.ts — owns session behavior independently of React. Prompt submission is serialized and asynchronous work revalidates the project, file, editor, and code epoch. Clear, resume, reconnect, and disposal invalidate stale work. Conversation lookup and mutations are included in awaited teardown.
4. ZookeeperFileRequestProcessor.ts and ZookeeperEditPatchHistory.ts — extracted file-integrity work. Undispatched stale requests are dropped; already-dispatched writes may settle, but cannot refresh or navigate a replacement project/editor. Undo history waits for both the filesystem write and the terminal exchange.
5. ZookeeperConversationPaneWrapper.tsx and ZookeeperConversationPane.tsx — read these last. Their large deletion is intentional: mounting and unmounting now only affects rendering. The outlet keeps a closeable loading shell visible while the controller or lazy pane is starting.
6. zookeeperManagerMachine.ts, SystemIO, and layout — small lifecycle seams for explicit socket shutdown, abortable idle waits, stale-navigation prevention, the running indicator, and the close/reopen E2E probe.

## Non-obvious behavior and review focus

- Every eligible project opens a Zookeeper connection and restores its saved conversation while the plugin is enabled, even if the pane has never opened. The React pane bundle remains lazy.
- Idle connection does not count as billing usage; usage remains tied to a prompt awaiting a response.
- Switching files or republishing the same path and project ID with the same editor preserves the actor and connection. Changing path, project ID, or editor creates a fresh session.
- Plugin disable/re-enable and replacement sessions that reuse a path or editor wait for pending work and persistence to drain. This favors data integrity, but a SystemIO operation that never invokes either filesystem completion callback could still block restart.
- A dispatched filesystem edit may finish safely after teardown so it is not left half-accounted-for, while stale UI navigation and editor updates are suppressed. The requestedProjectPath guard is the critical last line of defense here.

The highest-value lifecycle coverage is the Playwright happy path, which pauses an active response, closes the pane, lets the response finish while hidden, verifies a single WebSocket connection and the closed-pane spinner, then reopens the completed conversation and checks the resulting model. Unit coverage exercises eager activation, project identity, rapid toggles, queue/reset races, ordered persistence, asynchronous disposal, snapshot-to-patch undo fallback, and stale-project navigation suppression. Plugin disable/re-enable is not covered as a whole-app flow.

### CODE REVIEW

File: src/lib/layout/components.tsx [hunk 1]
```diff
@@ -530,7 +533,11 @@ function PaneButton({
         style={{ [buttonBorderWidthProp]: '2px' }}
         data-testid={`${pane.id}-pane-button`}
       >
-        <CustomIcon name={pane.icon} className="w-5 h-5" aria-hidden />
+        <CustomIcon
+          name={icon}
+          className={`w-5 h-5 ${icon === 'loading' ? 'animate-spin' : ''}`}
+          aria-hidden
+        />
```
Comment (pierremtb) [inline src/lib/layout/components.tsx:540]: Might as well :)

File: src/lib/zookeeper/components/ZookeeperConversation.spec.tsx [hunk 2]
```diff
@@ -1281,6 +1281,48 @@ describe('ZookeeperConversation', () => {
       ).not.toBeInTheDocument()
     })
 
+    test('closes screenshot annotation when the pane is hidden', () => {
+      const conversationProps = {
```
Comment (pierremtb) [inline src/lib/zookeeper/components/ZookeeperConversation.spec.tsx]: From Codex:

It’s a regression test for a portal-specific edge case.

Because #13142 keeps the Zookeeper React subtree mounted when its pane closes, the normal unmount cleanup no longer runs. If Zoodle screenshot annotation is active, its overlay is portaled into the viewport and would remain visible after closing Zookeeper.

The test verifies that changing `isPaneVisible` to `false` explicitly shuts down that overlay via [ZookeeperConversation.tsx](/Users/pierremtb/Documents/zoo/modeling-app/src/lib/zookeeper/components/ZookeeperConversation.tsx:309).

The duplicated prop fixture is ugly and could be cleaned up, but the behavior itself is necessary while we retain the portal approach.

File: src/lib/zookeeper/components/ZookeeperConversationPaneWrapper.tsx [hunk 3]
```diff
@@ -167,6 +158,11 @@ function ZookeeperConversationPaneInner(props: AreaTypeComponentProps) {
     zookeeperManagerActor,
```
Comment (pierremtb) [inline src/lib/zookeeper/components/ZookeeperConversationPaneWrapper.tsx]: This is where we better not mess up 😬

File: src/lib/zookeeper/registry/runtime.tsx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+) {
+  return (
+    left?.apiToken === right?.apiToken &&
+    left?.projectPath === right?.projectPath
+  )
+}
+
+// The registry owns the actor and transport. The portal remains a temporary
+// bridge for the React-based file, history, reconnect, queue, and billing hooks.
+export function createZookeeperRuntime(
```
Comment (pierremtb) [inline src/lib/zookeeper/registry/runtime.tsx]: I think this part from Codex is too complex and I'd like to make it easier to follow before marking ready.

**Review by pierremtb — COMMENTED** [review]
Some notes about the current state.

**Comment by pierremtb** [comment]
Superseded by the smaller two-PR stack: #13741 contains the runtime ownership and pane-independent lifetime change; #13743 contains the closed-pane activity indicator.

**Comment by pierremtb** [comment]
Closed in favor of this stack: https://github.com/KittyCAD/modeling-app/pull/13741
